src/lando/main/models/repo.py

```diff
@@ -38,6 +39,47 @@ class RepoError(Exception):
 class Repo(BaseModel):
     """Represents the configuration of a particular repo."""
 
+    path: str  # The path to the repository.
+    vcs: AbstractVcs
+
+    def push(
+        self, push_path: str, target: Optional[str] = None, force_push: bool = False
+    ):
+        return self.vcs.push(push_path, target, force_push)
+
+    def last_commit_for_path(self, repo_path: str, path: str) -> bytes:
+        return self.last_commit_for_path(repo_path, path)
+
+    @property
+    def push_target(self):
+        """The target branch or bookmark to push to."""
+        return self.push_bookmark or None
+
+    def apply_patch(self, patch_buf: StringIO):
+        return self.vcs.apply_patch(patch_buf)
+
+    def for_push(self, requester_email: str):
+        return self.vcs.for_push(requester_email)
+
+    @property
+    def revision_id(self):
+        return self.vcs.revision_id
+
+    def read_checkout_file(self, checkout_file):
+        return self.vcs.read_checkout_file(checkout_file)
+
+    def head_ref(self) -> str:
+        return self.vcs.head_ref()
+
+    def changeset_descriptions(self) -> list[str]:
+        return self.vcs.changeset_descriptions()
+
+    def update_repo(self, pull_path, target_cset):
+        return self.vcs.update_repo(pull_path, target_cset)
+
+    def format_stack(self, stack_size: int, bug_ids: list[str]):
+        return self.vcs.format_stack(stack_size, bug_ids)
```

Comment:
We should try and avoid this re-definition of `self.vcs` methods directly on the `Repo` class. As mentioned in our chat, we could do this with a method like `get_vcs` on `Repo` which returns a concrete object implementing `AbstractVcs`.